src/utils/markdown.ts

```diff
@@ -62,3 +62,27 @@ export const getHeadingLines = (
 	}
 	return { headingLines, minHeading, maxHeading };
 };
+
+// goes backwards from the `from` line, returns line number of first line containing any heading
+export const getPreviousHeading = (
+	editor: {
+		getLine: (number: number) => string;
+	}, 
+	from: number 
+) => {
+
+	let fence: FenceType = null;
+	let start = from > 0 ? from-- : 0
```

Comment:
Post decrement is not good because it does not affect the return value.
This would cause the `from` line to refer to itself when it contains a heading (repeating DeeperLevel will increase it permanently).

```diff
	let fence: FenceType = null;
-	let start = from > 0 ? from-- : 0
+	const start =  from > 0 ? from - 1 : 0;
```

```test.ts
test("'from line' contains heading", () => {
	const input = `# Heading1

## Heading2

Normal
`;
	const editor = new Editor(input);
	expect(getPreviousHeading(editor, 2)).toEqual(0);
});
```


Conversely, when `from` is 0 (the beginning of the file), it seems natural to base it on itself.

but I can't say for sure which one is better, so if you think this one is better, please modify it.

```diff
+  if(from===0) return 0;

	let fence: FenceType = null;
-	const start =  from > 0 ? from - 1 : 0;
+	const start = from - 1;
```

Reply:
Oooh, good catch, thanks! 👍 

Regarding your second point - I agree, it seems natural to return 0 here, but I think when looking at the method `getPreviousHeading` in isolation, I would expect it to only return a number if there is a heading found, otherwise `undefined`. With your suggestion, we would always return 0 when `from` is 0, even if no heading is present.